### ISSUE TITLE
🚀 - Add Shade of Purple theme

### ISSUE THREAD

**Issue by riccardoperra** [riccardoperra/codeimage#365]
### Which @codeimage/* package(s) are relevant/releated to the feature request?

highlight

### Description

This task consist of creating a new theme for CodeImage editor using CodeMirror API.

https://github.com/ahmadawais/shades-of-purple-vscode

---

## Instructions for contributions
- Follow the [CONTRIBUTING.md](https://github.com/riccardoperra/codeimage/blob/main/CONTRIBUTING.md) instructions to prepare your environment. 
> **NOTE** If your intention is to only develop a new theme, you don't need to use docker or install everything. You will work only on [@codeimage/highlight](https://github.com/riccardoperra/codeimage/tree/main/packages/highlight) package
- Move to [packages/highlight](https://github.com/riccardoperra/codeimage/tree/main/packages/highlight). You will see a [How to create a new theme section](https://github.com/riccardoperra/codeimage/tree/main/packages/highlight) in the README.md file .
- Run the `pnpm generate:theme` script and put the `shadeOfPurple` theme name
- Open the theme playground with `pnpm dev` (from the highlight package) or `pnpm dev --filter=@codeimage/highlight` (from the root folder)

For further help you can ask for help in the comments below

**Comment by SIMRAN88** [comment]
Hey @riccardoperra Can I work on this?

**Comment by SIMRAN88** [comment]
@riccardoperra When I run `pnpm dev` in the highlight folder, I get this error:  
<img width="1440" alt="Screenshot 2022-10-06 at 4 54 31 PM" src="https://user-images.githubusercontent.com/12448024/194300807-2c9fd0cd-7ab1-43e3-ad4d-5a1cce01fdf3.png">
Can you please help with this?

**Comment by riccardoperra** [comment]
Hey @SIMRAN88, have you launched the `pnpm libs:build` command from the root folder? It seems that it doesn't find the @codeimage/vanilla-extract dist folder.

Otherwise you can launch these two commands
```bash
cd packages/ui
pnpm build
```

```bash
cd packages/vanilla-extract
pnpm build  
```

**Comment by SIMRAN88** [comment]
Hey @riccardoperra , some error when I try to build the vanilla-extract
<img width="1440" alt="Screenshot 2022-10-06 at 5 16 39 PM" src="https://user-images.githubusercontent.com/12448024/194304790-5121bc1a-2299-434b-becf-d83cc11b7452.png">

The ui build has happened successfully
<img width="1440" alt="Screenshot 2022-10-06 at 5 18 11 PM" src="https://user-images.githubusercontent.com/12448024/194305026-c8fcdd23-94bd-4b3d-9d30-bbdc0770456b.png">

**Comment by SIMRAN88** [comment]
Hey @riccardoperra Thank you, it worked
<img width="1440" alt="Screenshot 2022-10-06 at 5 42 36 PM" src="https://user-images.githubusercontent.com/12448024/194309619-7fc8cb12-859a-4d0f-9bac-a08d3e742212.png">
This seems to look like this but the shadesOfPurple theme in vscode looks a little different.
something like this: 
<img width="1440" alt="Screenshot 2022-10-06 at 5 44 39 PM" src="https://user-images.githubusercontent.com/12448024/194309889-8330f41d-ce67-46a5-91fd-cd3291500d60.png">
 Any thoughts on this?

**Comment by riccardoperra** [comment]
Hey, the generated theme is just a placeholder 😅 it doesn’t automatically generate the theme from vscode.

there should be a vscode theme .json file with all the tokens which should be added In the CodeImage theme

**Comment by SIMRAN88** [comment]
> Hey, the generated theme is just a placeholder 😅 it doesn’t automatically generate the theme from vscode.
> 
> there should be a vscode theme .json file with all the tokens which should be added In the CodeImage theme
> 
> https://github.com/ahmadawais/shades-of-purple-vscode/blob/master/themes/shades-of-purple-color-theme.json
> 
> Obviously you don't have to port all the 300 tokens but just the most important ones present in the `defineEditorTheme` function (there are also other properties you can pass but they are optional)

Oops, my bad! 😅  I will make sure to follow the reference for the same. Thank you @riccardoperra

**Comment by riccardoperra** [comment]
resolved #370
